PWGCF/TwoParticleCorrelations/Tasks/pidDiHadron.cxx

```diff
@@ -55,17 +57,108 @@ using namespace o2::framework::expressions;
 // define the filtered collisions and tracks
 #define O2_DEFINE_CONFIGURABLE(NAME, TYPE, DEFAULT, HELP) Configurable<TYPE> NAME{#NAME, DEFAULT, HELP};
 
+template <typename T>
+auto projectMatrix(Array2D<T> const& mat, std::array<float, 6>& array1, std::array<float, 6>& array2, std::array<float, 6>& array3)
+{
+  for (auto j = 0; j < static_cast<int>(mat.cols); ++j) {
+    array1[j] = mat(0, j);
+    array2[j] = mat(1, j);
+    array3[j] = mat(2, j);
+  }
+  return;
+}
+template <typename T, typename P>
+auto readMatrix(Array2D<T> const& mat, P& array)
+{
+  for (auto i = 0; i < static_cast<int>(mat.rows); ++i) {
+    for (auto j = 0; j < static_cast<int>(mat.cols); ++j) {
+      array[i][j] = mat(i, j);
+    }
+  }
+
+  return;
+}
```

Comment:
This is really dangerous
The vectors structures does not have checks on their range which means that if they read or written out of their bounds most of the times it will not be noticed but other data is being read or overwritten which is really hard to fix and produces random results
Will it not work passing references to empty arrays of the right type and then filling the array with the corresponding elements?
Also named constants can be used to directly access the elements of the matrix without the need of serializing its content